catalyst-toolbox/src/rewards/veterans.rs

```diff
@@ -25,18 +30,34 @@ pub type EligibilityThresholds = std::ops::RangeInclusive<usize>;
 // e.g. something like an expanded version of a AdvisorReviewRow
 // [proposal_id, advisor, ratings, ..(other fields from AdvisorReviewRow).., ranking (good/excellent/filtered out), vca]
 
-fn calc_final_ranking_per_review(rankings: &[impl Borrow<VeteranRankingRow>]) -> ReviewRanking {
+// note that the consensus returned here is either FO/total or ((E+G)/total) because for now we do
+// not discriminate between Excellent and Good...
+fn calc_final_ranking_with_consensus_per_review(rankings: &[impl Borrow<VeteranRankingRow>]) -> FinalRankingWithConsensus {
     let rankings_majority = Decimal::from(rankings.len()) / Decimal::from(2);
     let ranks = rankings.iter().counts_by(|r| r.borrow().score());
 
-    match (ranks.get(&FilteredOut), ranks.get(&Excellent)) {
-        (Some(filtered_out), _) if Decimal::from(*filtered_out) >= rankings_majority => {
-            ReviewRanking::FilteredOut
+    match (ranks.get(&Excellent), ranks.get(&Good), ranks.get(&FilteredOut)) {
+        (_, _, Some(filtered_out)) if Decimal::from(*filtered_out) >= rankings_majority => {
+            FinalRankingWithConsensus {
+                review_ranking: FilteredOut,
+                consensus: Decimal::from(*filtered_out) / Decimal::from(rankings.len()),
+            }
         }
-        (_, Some(excellent)) if Decimal::from(*excellent) > rankings_majority => {
-            ReviewRanking::Excellent
+        (Some(excellent), maybe_good, _) if Decimal::from(*excellent) > rankings_majority => {
+            FinalRankingWithConsensus {
+                review_ranking: Excellent,
+                consensus: (maybe_good.map(|good| Decimal::from(*good)).unwrap_or_default()
+                    + Decimal::from(*excellent)) / Decimal::from(rankings.len()),
```

Comment:
In a way this interpretation is a bit different from what I would expect. We are promoting `good` votes to excellent as far as consensus goes but those were for a strictly lower level (i.e. if I voted for it to be good probably I think it's not worth of an excellent vote).
The opposite is fine ofc, if I voted for it to be excellent I would obviously be fine with it being good.

It's fine for me as long as everybody agrees on this, and if we go this way we might as well collapse these 2 match arms together

Reply:
I agree with you on the principle but I did it for consistency given that we do not discriminate between `Good` and `Excellent` anywhere else in the code, not doing this would keep `FO` ratings that should not be kept when the final rating is `Excellent`:

Since the current intent behind the `consensus` value is to prevent removing disagreeing ratings in a tie, only counting the `Excellents` to compute the consensus of an `Excellent` rating would decrease the consensus of the overall positivity of the `Excellent` rating and prevent excluding the FO ratings on those `Excellents`.

For example, in the case of a (4,1,1) we would have a consensus of 4/6 (<.70) which means that he vCA who chose `FO` would still be rewarded for their `FO` rating (if minimum consensus is set to 70%) whereas if we include the `Goods` in the consensus we get a consensus of 5/6 (0.83) and then the `FO` rating is no longer legitimate anymore (the `final_ranking_with_consensus.consensus < minimum_consensus` condition is `false` and the assessment is not kept).

This is what I've tried to explain in the comment above the function but if we wanted to be thorough the 'consensus' value should be renamed to something like `conseus_on_negativity_or_positivity` (maybe I should do it?). It's true that for now we just test for `.is_positive()` but this might change in the future so I think it's important to keep the two match arms.

Reply:
I understand your point, it just feels like there are a lot of very particular rules which make understanding the system as a whole more difficult than it should IMO (this is not on you ofc)

Reply:
I've renamed `consensus` to `negative_or_positive_consensus` and documented it in b4e3b49